src/components/Editor/Runners/PythonRunner/PyodideRunner/PyodideRunner.jsx

```diff
@@ -79,13 +80,20 @@ const PyodideRunner = () => {
   }, []);
 
   useEffect(() => {
-    if (codeRunTriggered) {
+    if (codeRunTriggered && active) {
+      console.log("running with pyodide");
       handleRun();
     }
   }, [codeRunTriggered]);
 
   useEffect(() => {
-    if (codeRunStopped) {
+    if (codeRunTriggered) {
+      setShowRunner(active);
+    }
+  }, [codeRunTriggered]);
```

Comment:
❓ is it `active` you want to check in these or `showRunner`? You might end up in a bit of a race condition but that's already potentially true anyway if `active` is toggled to true. So it might be worth merging these useEffects up

Reply:
Ignore me, I'm being an idiot 😅